main/test/flix/experimental/Test.Dec.AssocType.flix

```diff
@@ -39,4 +39,26 @@ mod Test.Dec.AssocType {
 
     @test
     def testRecord01(): Bool =  testRecordHelper01().x == 'a'
+
+    class Iterable2[a] {
+        type Eff[a]: Bool
+        type Elem[a]: Type
+
+        pub def iterator(rh: Region[r], x: a): Iterator[Iterable2.Elem[a], Iterable2.Eff[a] and r, r] & Iterable2.Eff[a] and r
```

Comment:
Will you ultimately put `Eff` and `Elem` into the local scope?

Reply:
Yes. This is related to an issue where you have to fully qualify signatures inside their own class as well.

Reply:
Ah- and its not an easy fix?

Reply:
It's slightly complex since it mixes namespaces. `Iterator` and `Elem` should both be in scope inside `Iterator`. It's a problem we've already solved for companion namespaces for enums, I believe, so it's probably not a big change. In any case, it is orthogonal to this PR.